plugins/inputs/mongodb/mongodb_server.go

```diff
@@ -22,6 +22,56 @@ func (s *Server) getDefaultTags() map[string]string {
 	return tags
 }
 
+type OplogStats struct {
+	TimeDiff int64
+}
+
+func (s *Server) gatherOplogStats() *OplogStats {
+	stats := &OplogStats{}
+	localdb := s.Session.DB("local")
+	oplog_collection_name := ""
+
+	collection_names, err := localdb.CollectionNames()
```

Comment:
It seems like instead of checking for the existence of this items we should just try to get the data and if it's not there then we don't report anything.  This will result in less network calls and checking doesn't guarantee the data will be there later anyway due to possible changes.

How about we just try to get the entries for `oplog.rs` and if that fails we can try `oplog.$main`?

Reply:
Yeah, it makes sense, will be fixed.